File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03126/p03126_num1_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N, M;
    std::cin >> N >> M;

    // Create a vector to store the count of each food
    std::vector<int> count(M + 1, 0);

    // Read the input and update the count
    for (int i = 0; i < N; i++) {
        int K;
        std::cin >> K;
        for (int j = 0; j < K; j++) {
            int food;
            std::cin >> food;
            count[food]++;
        }
    }

    // Count the number of foods liked by all people
    int likedByAll = 0;
    for (int i = 1; i <= M; i++) {
        if (count[i] == N) {
            likedByAll++;
        }
    }

    std::cout << likedByAll << std::endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03126/p03126_num1_parsed.py
N, M = map(int, input().split())
liked_count = [0] * M

for _ in range(N):
    foods = list(map(int, input().split()))[1:]
    for food in foods:
        liked_count[food - 1] += 1

result = liked_count.count(N)
print(result)